Sources/Apollo/ApolloErrorInterceptor.swift

```diff
@@ -0,0 +1,20 @@
+import Foundation
+
+/// An error interceptor called to allow further examination of error data when an error occurs in the chain.
+public protocol ApolloErrorInterceptor {
```

Comment:
I'm not sure if it would work, but what do you think of instead using protocol for this interceptor, actually, use just a function. For instance, 

`typealias ApolloErrorInterceptor =  (Error,
       RequestChain,
       HTTPRequest<Operation>,
       HTTPResponse<Operation>?,
       @escaping (Result<GraphQLResult<Operation.Data>, Error>) -> Void)) -> Void`

Reply:
I would wayyyyyy rather have this as a protocol, particularly since it gives longer term flexibility to add more methods if needed.